probe-rs/src/architecture/arm/memory/romtable.rs

```diff
@@ -111,11 +111,14 @@ impl RomTable {
     /// This does not check whether the data actually signalizes
     /// to contain a ROM table but assumes this was checked beforehand.
     fn try_parse(memory: &mut Memory<'_>, base_address: u64) -> Result<RomTable, RomTableError> {
+        // This is not a needless collect! It fixes the borrowing issue with &mut Memory!
```

Comment:
Which collect is the issue here? I think it would be better if the comment is at the same place where the call to `collect` is.